Keep the names of the variables with the table

avdg81 opened the pull request:
**📝 Description**
The names of the variables X and Y of a table were read (from the `mdpa` file), but not stored. Now they are. This allows the user of a table to give meaning to the stored values and act differently if needed.

**🆕 Changelog**
- Added (private) data members to keep the names of X and Y with the table. Also added getters and setters for these.
- When a table is being read from an `mdpa` file, no longer ignore the names of X and Y, but store them.
- Removed some commented out code.
- Use compiler-generated special member functions (e.g. copy constructors) rather than reimplementing the default behavior.
- Single-argument constructors should be `explicit`.
- Use nested namespace definitions to improve readability.
- Removed needless indentation and some redundant comments.
- Added a unit test to cover the new functionality.

avdg81 committed:
Keep the names of the variables with the table

The names of the variables X and Y of a table were read (from the `mdpa`
file), but not stored. Now they are. This allows the user of a table to
give meaning to the stored values and act differently if needed.

Other changes include:
- Removed some commented out code.
- Use compiler-generated special member functions (e.g. copy
  constructors) rather than reimplementing the default behavior.
- Single-argument constructors should be `explicit`.
- Use nested namespace definitions to improve readability.
- Removed needless indentation and some redundant comments.

loumalouomega reviewed (changes requested):
Blocking to avoid accidental merge

roigcarlo committed:
Fix test macro conflict

loumalouomega reviewed (dismissed):
I remove my block, but wait until @KratosMultiphysics/technical-committee agrees

loumalouomega committed:
Fix compilation

philbucher commented on kratos/tests/cpp_tests/sources/test_table.cpp:
```diff
@@ -5,10 +5,10 @@
 //
 //  Main authors:    Vicente Mataix Ferrandiz
-//
+//                   Anne van de Graaf
```
```suggestion
```

[FYI](https://github.com/KratosMultiphysics/Kratos/wiki/Authorship-of-Kratos-files)

avdg81 committed:
Reverted unintended adding of a main author

Co-authored-by: Philipp Bucher <philipp.bucher@tum.de>

pooyan-dadvand reviewed (dismissed):
Thank you for the implementation and also your clear explanation!

avdg81 committed:
Processed review feedback

Added the names for variables X and Y to the `Table` class template too,
not just to the template specialization `Table<double, double>`.

pooyan-dadvand reviewed (approved):
Thank you again!
